shared/fs/header/header.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../../styles'
+import {Avatar, Box, ClickableBox, Icon, Text} from '../../common-adapters'
+import HeaderConnector from './header-container'
+
+type FolderHeaderProps = {
+  breadcrumbItems: Array<Types.PathBreadcrumbItem>,
+  dropdownItems: Array<Types.PathBreadcrumbItem>,
+  isTeamPath: boolean,
+  onOpenBreadcrumb: (path: string) => void,
+  onOpenBreadcrumbDropdown: (
+    dropdownItems: Array<Types.PathBreadcrumbItem>,
+    isTeamPath: boolean,
+    onOpenBreadcrumb: (path: string) => void,
+    targetRect: ?ClientRect
+  ) => void,
+}
+
+const FolderHeader = HeaderConnector(
+  ({
+    dropdownItems,
+    breadcrumbItems,
+    isTeamPath,
+    onOpenBreadcrumb,
+    onOpenBreadcrumbDropdown,
+  }: FolderHeaderProps) => (
+    <Box>
+      {breadcrumbItems.length === 1 ? (
+        <Box style={folderHeaderStyleRoot}>
+          <Text type="BodyBig">Keybase Files</Text>
+        </Box>
+      ) : (
+        <Box style={folderHeaderStyleTree}>
+          {dropdownItems.length > 0 && (
+            <Box style={folderBreadcrumbStyle}>
+              <ClickableBox
+                style={styleBreadcrumbDropdownIconBox}
+                onClick={evt => {
```

Comment:
when this expands to include more parts of the design, it'll probably make sense to expand this to a class & factor these out. should also make it cleaner to deal with the mobile case switching like Nojima mentioned

Reply:
Do you have an example for what you're describing?

Reply:
Something like:

```
class Header extends React.Component<Props> {

  const _onClick = (evt) => {
    <longer fcn containing bunch of checks>
  }

  const _mapBreadcrumbs = (i, idx) => {
    <fcn returning constructed node>
  }

  render() {
    ...
    <ClickableBox onClick={this._onClick}> ... </ClickableBox>
    ...
    {breadCrumbItems.map(this._mapBreadcrumbs)}
    ...
  }
}

Reply:
Gotcha. Yeah I can do that eventually.